File: data/readable_format/fortran/file692.f90
module orders
  implicit none
contains
  ! Implementation of lt_order for double precision
  logical function lt_dble(a, b)
    double precision, intent(in) :: a, b
    lt_dble = a < b
  end function lt_dble

  ! Implementation of lt_order for integer
  logical function lt_int(a, b)
    integer, intent(in) :: a, b
    lt_int = a < b
  end function lt_int

  ! Implementation of lt_order for real
  logical function lt_real(a, b)
    real, intent(in) :: a, b
    lt_real = a < b
  end function lt_real

  ! Add similar direct implementations for gtabs_order and gt_order
  ! as needed for your specific requirements
end module orders

program test_orders
  use orders
  implicit none

  ! Test variables
  real :: r1 = 1.0, r2 = 2.0
  integer :: i1 = 1, i2 = 2
  double precision :: d1 = 1.0d0, d2 = 2.0d0
  logical :: testPassed

  ! Testing lt_order
  testPassed = lt_dble(d1, d2) .and. lt_int(i1, i2) .and. lt_real(r1, r2)
  if (testPassed) then
    print *, "lt_order tests passed."
  else
    print *, "lt_order tests failed."
  end if

  ! Add tests for gtabs_order and gt_order as needed
end program test_orders

File: data/readable_format/cpp/file692.cpp
#include <cmath> // For std::abs
#include <iostream> // For std::cout

class Orders {
public:
    // Overloaded lt_order (Less Than) functions
    static bool lt_order(double a, double b) {
        return a < b;
    }
    static bool lt_order(int a, int b) {
        return a < b;
    }
    static bool lt_order(float a, float b) {
        return a < b;
    }

    // Overloaded gtabs_order (Greater Than Absolute) functions
    static bool gtabs_order(double a, double b) {
        return std::abs(a) > std::abs(b);
    }
    static bool gtabs_order(int a, int b) {
        return std::abs(a) > std::abs(b);
    }
    static bool gtabs_order(float a, float b) {
        return std::abs(a) > std::abs(b);
    }

    // Overloaded gt_order (Greater Than) functions
    static bool gt_order(double a, double b) {
        return a > b;
    }
    static bool gt_order(int a, int b) {
        return a > b;
    }
    static bool gt_order(float a, float b) {
        return a > b;
    }
};

// Main function for testing
int main() {
    // Testing lt_order
    if (Orders::lt_order(1, 2) && Orders::lt_order(1.0, 2.0) && Orders::lt_order(1.0f, 2.0f)) {
        std::cout << "lt_order tests passed." << std::endl;
    } else {
        std::cout << "lt_order tests failed." << std::endl;
    }

    // Testing gtabs_order
    if (Orders::gtabs_order(-2, 1) && Orders::gtabs_order(-2.0, 1.0) && Orders::gtabs_order(-2.0f, 1.0f)) {
        std::cout << "gtabs_order tests passed." << std::endl;
    } else {
        std::cout << "gtabs_order tests failed." << std::endl;
    }

    // Testing gt_order
    if (!Orders::gt_order(1, 2) && !Orders::gt_order(1.0, 2.0) && !Orders::gt_order(1.0f, 2.0f)) {
        std::cout << "gt_order tests passed." << std::endl;
    } else {
        std::cout << "gt_order tests failed." << std::endl;
    }

    return 0;
}